src/site/controllers/StaticResourcesController.php

```diff
@@ -54,24 +55,23 @@ protected static function getExtraParametersSpec(
       throw new HTTPNotFoundException();
     }
 
-    await $response->getBody()
-      ->writeAsync(\file_get_contents($entry['localPath']));
-    $response = $response->withHeader('Content-Type', vec[$entry['mimeType']]);
+    $response = response(
+      Message\StatusCode::Ok,
+      dict['Content-Type' => vec[$entry['mimeType']]],
+      Body\file($entry['localPath']),
+    );
 
     if (
       $checksum === 'local-changes' && $this->getParameters()['MTime'] === null
     ) {
-      $response = $response->withAddedHeader(
-        'Cache-Control',
-        vec['max-age=0, no-cache, no-store'],
-      );
+      $response = Response\with_max_age($response, 0)
+        |> Response\with_cache_control_directive($$, 'no-cache')
+        |> Response\with_cache_control_directive($$, 'no-store');
     } else {
-      $response = $response->withAddedHeader(
-        'Cache-Control',
-        vec['max-age=31556926'], // 1 year
-      )
-        ->withAddedHeader('ETag', vec['"'.$checksum.'"']);
+      $response = Response\with_max_age($response, 31556926) // 1 year
+        |> Response\with_etag($$, $checksum);
```

Comment:
no need to do `'"'.$checksum.'"'`, it will be done automatically if the checksum is not wrapped within `"`